oeps/best-practices/oep-0066-bp-openfeature-toggles.rst

```diff
@@ -0,0 +1,124 @@
+.. _pep_based_template:
+
+.. Below is the display in the left sidebar on RTD. Please omit leading 0's
+
+OEP-66: OpenFeature Format For Feature Toggles
+##############################################
+
+.. This OEP template is based on Python's PEP standard.
+
+.. list-table::
+   :widths: 25 75
+
+   * - OEP
+     - Link to the doc in the following format::
+
+        :doc:`OEP-0066 <oep-0066-bp-openfeature-toggles>`
+
+   * - Title
+     - Implement All Toggles Using OpenFeature Specification
+   * - Last Modified
+     - 2025-01-15
+   * - Authors
+     - Tobias Macey
+   * - Arbiter
+     - Feanil Patel <feanil@axim.org>
+   * - Status
+     - Draft
+   * - Type
+     - Best Practice
+   * - Created
+     - 2025-01-15
+   * - Review Period
+     - 2025-01-15 - 2025-02-28
+   * - Resolution
+     - <links to any discussions where the final status was decided>
+   * - References
+     - `OEP 0017 <https://open-edx-proposals.readthedocs.io/en/latest/best-practices/oep-0017-bp-feature-toggles.html>`_
+
+Abstract
+********
+
+OpenFeature is an open standard and ecosystem for managing feature toggles across
+various languages and frameworks. This offers a single, consistent format for managing
+the ubiquitous toggles in Open edX to improve their discoverability and maintenance.
+
+Motivation
+**********
+
+OEP 17 provided an excellent and valid argument in favor of the adoption of feature
+toggles as a core practice of the Open edX suite of software. Unfortunately, the
+implementation of that practice has led to a large variety in the methods used to manage
+those toggles, and dramatically different naming across repository boundaries. The
+adoption of the `OpenFeature <https://openfeature.dev/>`_ specification and associated
```

Comment:
Thanks. Some questions:
1. Is it expected that someone needs to read and become familiar with all of https://openfeature.dev/ as part of accepting this proposal, or is it possible to summarize what we are taking from the spec?
2. If the original OEP still stands, and this is just an extension of it, it still might make sense to add a note and link in the original toggle OEP.
3. OpenFeature has features like Hooks and Events. Are those compatible with chained providers? Would we need to explain in which ways these could and could not be used?
4. Does OpenFeature have details around how toggles should be annotated that would override any of the ADRs or How-Tos in edx-toggles? See https://edx.readthedocs.io/projects/edx-toggles/en/latest/readme.html.